src/haz3lcore/dynamics/Builtins.re

```diff
@@ -75,16 +110,51 @@ module Pervasives = {
         raise(EvaluatorError.Exception(InvalidBoxedFloatLit(d1)))
       | Indet(d1) => Indet(ApBuiltin(name, [d1])) |> return
       };
+
+    /* float_of_string implementation. */
+    let float_of_string = (name, r1) =>
+      switch (r1) {
+      | BoxedValue(StringLit(s) as d1) =>
+        let s = Re.Str.string_after(s, 1);
+        let s = Re.Str.string_before(s, String.length(s) - 1);
+        if (String.length(s) == 0) {
+          BoxedValue(FloatLit(0.)) |> return;
+        } else if (Form.is_float(s)) {
```

Comment:
maybe more efficient to just catch the exception that float_of_string would throw instead of separately checking if the input is a float